feat: introduce native utils getPublicKey polyfill

Nodonisko opened the pull request:
## **Description**

Replace JS `getPublicKey` function with our new C++ implementation. 

As you can see in videos it made login ~4s faster but I would take that with grain of salt as login time is quite variable due to other variables, it would require multiple measurements and averages and that would require quite some time and I plan to do it once we have all C++ function implemented in the app.

According to code profiling I expect it to be around ~2s for SRP with 100+ accounts.

<img width="293" height="209" alt="Screenshot 2025-09-25 at 23 40 16" src="https://github.com/user-attachments/assets/875d43a1-3265-4435-9c43-abe060cf3e4f" />
<img width="271" height="197" alt="Screenshot 2025-09-25 at 23 40 23" src="https://github.com/user-attachments/assets/0088b7ee-0b97-4544-ab92-aec7af77181a" />



CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Just general smoke test, login, account import, SRP import...

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**


https://github.com/user-attachments/assets/7250fee7-b111-4b46-87a5-e5638606a7b2



### **After**


https://github.com/user-attachments/assets/d8071433-5392-467a-ac97-e63ce8fe39aa





## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Replaces JS secp256k1 getPublicKey with a native C++ polyfill and refactors Ramp payment method selection into a new bottom-sheet modal with routing/tests, plus required iOS/dep updates.
> 
> - **Native/Crypto**:
>   - Polyfill `@noble/curves` `secp256k1.getPublicKey` with native `@metamask/native-utils` in `shim.js`.
>   - Add deps: `@metamask/native-utils`, `react-native-nitro-modules`, pin `@noble/curves@1.9.6`, `@scure/bip32@1.7.0`; update `yarn.lock`.
>   - iOS: add pods `NitroModules`, `NativeUtils`; update `ffi` in `Gemfile.lock`.
> - **Ramp UI (Buy/Sell)**:
>   - Replace `PaymentMethodModal` with new bottom-sheet `PaymentMethodSelectorModal` (styles, tests, snapshots).
>   - Update `BuildQuote` to navigate to `RampPaymentMethodSelectorModal` instead of inline modal; adjust handlers.
>   - Add route `Routes.RAMP.MODALS.PAYMENT_METHOD_SELECTOR` and register in `routes/index.tsx`.
> - **Tests**:
>   - Update `BuildQuote.test.tsx` to assert navigation to selector modal; add dedicated tests for the new modal.
> - **Tooling**:
>   - Add `react-native-nitro-modules` to `.depcheckrc.yml` ignores.
>   - Minor formatting changes in analytics/events and network banner tracking.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b946d08ef34b582141d30869dcc5040c0616cb2f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Nodonisko committed:
feat: introduce native utils getPublicKey polyfill

ngalee8311 commented:
margelo:feat/introduce-native-utils
Files changed

Nodonisko committed:
fix: add missing dependency

Nodonisko commented:
Updated, rebased, fixed comments.

danroc reviewed (dismissed):
LGTM, I tested on a Pixel 7.

Nodonisko committed:
chore: add nitro modules to depcheck ignore

Nodonisko committed:
chore: improve comment a bit

georgeweiler committed:
refactor: ramp payment modal bottom sheet (#20562)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR refactors the ramp payment selector modal to use the bottom
sheet pattern which will remove the flashy UI that existed previously.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

`CHANGELOG entry: changed ramp payment method selector modal to a bottom
sheet`


## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**


https://github.com/user-attachments/assets/b32cc7e9-9f3a-42d6-b583-bfd91445a200


<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->


https://github.com/user-attachments/assets/a345e5bb-0370-4b9b-8f0f-b19f1e696fd7





## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Replaces in-view PaymentMethodModal with a navigated bottom-sheet
PaymentMethodSelectorModal and updates BuildQuote, routes, and tests
accordingly.
> 
> - **UI/Flow (Ramp)**:
> - Replace `PaymentMethodModal` with bottom-sheet
`PaymentMethodSelectorModal`.
> - In `BuildQuote`, payment method "Change" now navigates to
`RampModals/RampPaymentMethodSelectorModal` via
`createPaymentMethodSelectorModalNavigationDetails`.
> - Remove local modal state/handlers; add
`handleShowPaymentMethodsModal` navigation.
> - **Navigation**:
> - Add `Routes.RAMP.MODALS.PAYMENT_METHOD_SELECTOR` and register in
`RampModalsRoutes`.
> - **Components**:
> - Add `components/PaymentMethodSelectorModal` (styles, index, tests,
snapshots).
>   - Delete legacy `components/PaymentMethodModal.tsx`.
> - **Tests**:
> - Update `BuildQuote.test.tsx` to expect navigation to selector modal
instead of direct selection.
> - Add unit tests and snapshots for `PaymentMethodSelectorModal`,
including analytics on selection.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
8ad1e908b9d0df06d9321c5c22cd3e92fe67591a. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

sethkfman reviewed (approved):
LGTM
